hera_cal/nucal.py

```diff
@@ -0,0 +1,464 @@
+
+        # Add baseline lengths to length dictionary
+        for bl in group:
+            ant1, ant2, _ = bl
+            blmag = np.linalg.norm(self.antpos[ant2] - self.antpos[ant1])
+            self.baseline_lengths[bl] = blmag
+
+        # Reset the group now that radially redundant groups have changed
+        self._reset_mapping_dictionaries()
+
+    def __len__(self):
+        """Get number of frequency redundant groups"""
+        return len(self._radial_groups)
+
+    def __getitem__(self, index):
+        """Get RadialRedundantGroup object from list of unique orientations"""
+        return self._radial_groups[index]
+
+    def __setitem__(self, index, value):
+        """
+        Set value at index in _radial_groups. Also raises an error if a baseline with the same
+        heading is already in the list of radially redundant groups
+        """
+        # Check to make sure the new group is radially redundant
+        self._check_new_group(value)
+            
+        for bl in self._mapped_spectral_reds:
+            if is_same_orientation(value[0], bl, self.antpos, blvec_error_tol=self.blvec_error_tol) and bl[-1] == value[0][-1]:
+                raise ValueError('Radially redundant group with same orientation and polarization already exists in the data')
+                
+        # Add group at index
+        self._radial_groups[index] = value
+
+        # Add baseline lengths to length dictionary
+        for bl in value:
+            ant1, ant2, _ = bl
+            blmag = np.linalg.norm(self.antpos[ant2] - self.antpos[ant1])
+            self.baseline_lengths[bl] = blmag
+
+        # Reset baseline mapping to spectrally redundant groups
+        self._reset_mapping_dictionaries()
+
+    def append(self, value):
+        """
+        Append value to the end of _radial_groups. Also raises an error if a baseline with the same
+        heading is already in the list of radially redundant groups
+        """
+        # Check to make sure the new group is radially redundant
+        self._check_new_group(value)
+        
+        for bl in self._mapped_spectral_reds:
+            if is_same_orientation(value[0], bl, self.antpos, blvec_error_tol=self.blvec_error_tol) and bl[-1] == value[0][-1]:
+                raise ValueError('Radially redundant group with same orientation and polarization already exists in the data')
+
+        # Append new group
+        self._radial_groups.append(value)
+
+        # Add baseline lengths to length dictionary
+        for bl in value:
+            ant1, ant2, _ = bl
+            blmag = np.linalg.norm(self.antpos[ant2] - self.antpos[ant1])
+            self.baseline_lengths[bl] = blmag
+
+        # Reset baseline mapping to spectrally redundant groups
+        self._reset_mapping_dictionaries()
+    
+    def __iter__(self):
+        """Iterates through the list of redundant groups"""
+        return iter(self._radial_groups)
+
+    def sort(self, key=None, reverse=True):
```

Comment:
The key argument is just ignored. Is that intentional?

`key=(len if key is None else key)` might work

Reply:
Yeah, good point. Just changed this to your suggestion